checkov/dockerfile/image_referencer/provider.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+import logging
+import os
+from typing import TYPE_CHECKING, Callable, Any
+
+from checkov.common.images.image_referencer import Image
+from checkov.common.util.str_utils import removeprefix
+from checkov.dockerfile.utils import DOCKERFILE_STARTLINE, DOCKERFILE_ENDLINE
+
+if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
+
+_ExtractImagesCallableAlias: TypeAlias = Callable[["dict[str, Any]"], "list[str]"]
+
+
+class DockerfileProvider:
+    __slots__ = ("definitions", "supported_resource_types")
+
+    def __init__(self, definitions: dict[str, Any]) -> None:
+        self.definitions = definitions
+
+    def extract_images_from_resources(self) -> list[Image]:
+        images = []
+
+        for file_path, config in self.definitions.items():
+            instructions = config.get("FROM")
+            if not isinstance(instructions, list):
+                continue
+
+            for instruction in instructions:
+                name = instruction["value"]
+
+                if " AS " in name:
+                    # indicates a multi-stage build, therefore remove everything starting from AS
+                    # ex. FROM maven:3.8-openjdk-17-slim AS build
+                    name = name.split(" AS ")[0]
+
+                if name.startswith("--platform"):
+                    # indicates a multi-platform build, therefore remove the platform flag and its value
+                    # ex. FROM --platform=$BUILDPLATFORM golang:alpine AS build
+                    multi_platform = name.split(" ", maxsplit=1)
+                    if len(multi_platform) != 2:
+                        logging.info(f"Invalid FROM statement {instruction['value']}")
+                        continue
+
+                    name = multi_platform[1]
+
+                images.append(
```

Comment:
@gruebel I thought we wanted to take only the last FROM as it is the base image. The rest may bring CVEs with them, but this image will definitely bring CVEs that need fixing...

Reply:
yeah, will adjust it 🙂